js/data/data.js

```diff
@@ -0,0 +1,124 @@
+var Firebase = require('firebase');
+var FirebaseUtils = require('../utils/firebase');
+
+// TODO put the uri in a constant / env var
+var FirebaseRef = new Firebase('https://quack.firebaseio.com');
+
+module.exports = {
+  create: function(resourceType, options) {
+    var handlers = {
+      message: function() {
+        if (!options || !options.message) {
```

Comment:
Log an error and return out if no options were specified and then you don't have to keep checking if options were provided.